packages/core/src/NodeExecuteFunctions.ts

```diff
@@ -1301,13 +1303,31 @@ export function returnJsonArray(jsonData: IDataObject | IDataObject[]): INodeExe
 		jsonData = [jsonData];
 	}
 
-	jsonData.forEach((data) => {
+	jsonData.forEach((data: IDataObject) => {
 		returnData.push({ json: data });
 	});
 
 	return returnData;
 }
 
+/**
+ * Takes generic input data and brings it into the new json, pairedItem format n8n uses.
+ * @export
+ * @param {(IPairedItemData)} itemData
+ * @param {(INodeExecutionData[])} inputData
+ * @returns {(INodeExecutionMetaData[])}
+ */
+export function constructExecutionMetaData(
+	itemData: IPairedItemData | IPairedItemData[],
+	inputData: INodeExecutionData[],
+): INodeExecutionMetaData[] {
+	return inputData.map((data: INodeExecutionData) => {
+		const { json, ...rest } = data;
+		const metaData = { json, pairedItem: { ...itemData }, rest } as INodeExecutionMetaData;
+		return metaData;
+	});
```

Comment:
We should spread the `rest` variable - in fact, we should be able to simplifly this block to

```
	return inputData.map((data: INodeExecutionData) => {
		return { pairedItem: { ...itemData }, ...data };
	});
```

Reply:
I think readability with json being explicitly extracted and show might be a benefit no?